src/PREfastXmlSarifConverter/DefectXmlFormat.h

```diff
@@ -38,11 +38,17 @@ class XmlKeyEvent
     {
     }
 
-    const wchar_t* GetId() const { return m_id.c_str();  }
-    const wchar_t* GetKind() const { return m_kind.c_str(); }
-    const wchar_t* GetImportance() const { return m_importance.c_str(); }
-    const wchar_t* GetMessage() const { return m_message.c_str(); }
-    bool IsValid() const { return m_valid; }
+	const wchar_t* GetId() const { return m_id.c_str(); }
+	const wchar_t* GetKind() const { return m_kind.c_str(); }
+	const wchar_t* GetImportance() const { return m_importance.c_str(); }
+	const wchar_t* GetMessage() const { return m_message.c_str(); }
+	bool IsValid() const { return m_valid; }
+
+    void SetId(const wchar_t *id) { m_id = id;  }
+    void SetKind(const wchar_t *kind) { m_kind = kind; }
```

Comment:
I am not strong it as a raw pointer. The values gets copied into std::wstring m_id object. 
I have tested the following code: 

{
    wchar_t *p = new wchar_t[100];
    wcscpy_s(p, 100, L"Foo");
    std::wstring m_str = p; 
    delete p;
    // m_str is valid here
}